src/Hazelcast.Net.Tests/Serialization/Compact/GenericRecordTests.cs

```diff
@@ -0,0 +1,410 @@
+            {
+                FieldKind.Compact => "GenericRecord",
+                FieldKind.ArrayOfCompact => "ArrayOfGenericRecord",
+                _ => kinds[i].ToString()
+            };
+            var m = typeof(CompactGenericRecordBuilder).GetMethod($"Set{kind}");
+            Assert.That(m, Is.Not.Null, $"Failed to get method CompactGenericRecordBuilder.Set{kinds[i]}");
+            var v = ReflectionDataSource.GetValueOfType(m!.GetParameters()[1].ParameterType);
+            var e = Assert.Throws<TargetInvocationException>(() => m.Invoke(recordBuilder, new[] { $"field-{i}", v }));
+            Assert.That(e.InnerException, Is.InstanceOf<SerializationException>());
+        }
+
+        // can build
+        var rec = recordBuilder.Build();
+
+        // can read all fields
+        for (var i = 1; i < kinds.Length; i++)
+        {
+            var kind = kinds[i] switch
+            {
+                FieldKind.Compact => "GenericRecord",
+                FieldKind.ArrayOfCompact => "ArrayOfGenericRecord",
+                _ => kinds[i].ToString()
+            };
+            var m = typeof(IGenericRecord).GetMethod($"Get{kind}");
+            Assert.That(m, Is.Not.Null, $"Failed to get method IGenericRecord.Get{kind}");
+            var ms = typeof(CompactGenericRecordBuilder).GetMethod($"Set{kind}");
+            Assert.That(ms, Is.Not.Null, $"Failed to get method CompactGenericRecordBuilder.Set{kind}");
+            var v = ReflectionDataSource.GetValueOfType(ms!.GetParameters()[1].ParameterType);
+            var r = m.Invoke(rec, new object[] { $"field-{i}" });
+            Assert.That(r, Is.EqualTo(v));
+        }
+    }
+
+    [Test]
+    public void DictionaryGenericRecordEdgeCase1()
+    {
+        // array of int can be obtained as array of int?
+
+        var schema = SchemaBuilder.For("test").WithField("field", FieldKind.ArrayOfInt32).Build();
+        var fieldValues = new Dictionary<string, object?>
+        {
+            {"field", new[] { 1, 2, 3 } }
+        };
+        var rec = new CompactDictionaryGenericRecord(schema, fieldValues);
+        var a = rec.GetArrayOfInt32("field");
+        Assert.That(a, Is.Not.Null);
+        Assert.That(a!.Length, Is.EqualTo(3));
+        var an = rec.GetArrayOfNullableInt32("field");
+        Assert.That(an, Is.Not.Null);
+        Assert.That(an!.Length, Is.EqualTo(3));
+    }
+
+    [Test]
+    public void DictionaryGenericRecordEdgeCase2()
+    {
+        // array of int? can be obtained as array of int
+
+        var schema = SchemaBuilder.For("test").WithField("field", FieldKind.ArrayOfNullableInt32).Build();
+        var fieldValues = new Dictionary<string, object?>
+        {
+            { "field", new int?[] { 1, 2, 3 } }
+        };
+        var rec = new CompactDictionaryGenericRecord(schema, fieldValues);
+        var a = rec.GetArrayOfInt32("field");
+        Assert.That(a, Is.Not.Null);
+        Assert.That(a!.Length, Is.EqualTo(3));
```

Comment:
Can we assert the values? I mean, what if we read it wrongly when int is read as int? ? Similar for the other tests.

Reply:
I did not mean the type will be different. I mean we can read the value wrongly somehow. Like reading 1 as 2 somehow. So can we  assert that a is `{ 1, 2, 3 }`

Reply:
ah I get it, fixing